### PR TITLE
Lon 961

### PR DESCRIPTION
**Pull request by SandraPG85**
** ScreenState extended to add a Freeze state that modal views call to instead of Hide.
** StateController extended to wait for a modal transition to be completed
** EventController merged with CoreEventsController
** Fix popup windows in dashboard events and substitution events from tree view

### CODE REVIEW

File: VAS.Core/Common/EventEditionSettings.cs [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+//  GNU General Public License for more details.
+//
+//  You should have received a copy of the GNU General Public License
+//  along with this program; if not, write to the Free Software
+//  Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA.
+//
+namespace VAS.Core.Common
+{
+	/// <summary>
+	/// Settings of the play events edition
```
Comment (vguzmanp) [inline VAS.Core/Common/EventEditionSettings.cs]: Can you give a bit more info about the class? Intended usage, etc

File: VAS.Core/Common/StateController.cs [hunk 2]
```diff
@@ -372,7 +383,7 @@ public bool UnRegister (string transition)
 				if (!await navigationStateStack [i].ScreenState.HideState ()) {
 					return false;
 				}
-				if (!await navigationStateStack [i].ScreenState.UnloadState ()) {
+				if (!await navigationStateStack [i].Unload ()) {
```
Comment (vguzmanp) [inline VAS.Core/Common/StateController.cs]: Why is this one different from the others? Shouldn't it be in ScreenState?

File: VAS.Services/Controller/CoreEventsController.cs [hunk 3]
```diff
@@ -38,59 +35,48 @@
 
 namespace VAS.Services.Controller
 {
-	// FIXME: Merge with the EventsController
```
Comment (vguzmanp) [inline VAS.Services/Controller/CoreEventsController.cs]: Is this actually fixed?

File: VAS.UI.Gtk2/UI/TreeView/TimelineEventsTreeView.cs [hunk 4]
```diff
@@ -139,6 +139,9 @@ protected override void HandleTreeviewSelectionChanged (object sender, EventArgs
 			if (Selection.CountSelectedRows () == 0) {
 				ViewModel.UnloadEvents ();
 			}
+
+			// update selection
+			ViewModel.FullTimeline.Selection.Replace (events);
```
Comment (vguzmanp) [inline VAS.UI.Gtk2/UI/TreeView/TimelineEventsTreeView.cs]: How is this related to the rest?

File: VAS.Services/Controller/EventsController.cs [hunk 5]
```diff
@@ -138,6 +293,101 @@ void HandleFiltersChanged ()
...
+		}
+
+		void HandlePlaylistElementLoaded (PlaylistElementLoadedEvent e)
+		{
+			if (e.Element is PlaylistPlayElement) {
+				LoadedPlay = (e.Element as PlaylistPlayElement).Play;
+			} else {
+				LoadedPlay = null;
+			}
+		}
```
Comment (vguzmanp) [inline VAS.Services/Controller/EventsController.cs]: From all these changes, what are movements from the other controller and what are actual changes?

File: VAS.Core/Interfaces/IEventEditorService.cs [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+using VAS.Core.Common;
+using VAS.Core.Store;
+using VAS.Core.ViewModel;
+
+namespace VAS.Core.Interfaces
+{
+	/// <summary>
+	/// Service that offers graphical event edition
+	/// </summary>
+	public interface IEventEditorService
```
Comment (vguzmanp) [inline VAS.Core/Interfaces/IEventEditorService.cs]: We have a `Service` interface, that works differently to this one. In general, services are application-wide controllers, but this one should be a normal controller, shouldn't it?

File: VAS.Core/Common/StateController.cs (hunk 2 above)
Comment (SandraPG85) [inline VAS.Core/Common/StateController.cs]: Notify about the completion of the navigation is responsibility of the navigation state instead of the screen state because is a more oriented in the navigation.

File: VAS.Services/Controller/CoreEventsController.cs (hunk 3 above)
Comment (SandraPG85) [inline VAS.Services/Controller/CoreEventsController.cs]: I checked with Andoni the events that can be moved to the events controller, the remaining step should be find a better name for the CoreEventsController

File: VAS.UI.Gtk2/UI/TreeView/TimelineEventsTreeView.cs (hunk 4 above)
Comment (SandraPG85) [inline VAS.UI.Gtk2/UI/TreeView/TimelineEventsTreeView.cs]: this fixes a bug in the selection, since it was all time empty.

File: VAS.Services/Controller/EventsController.cs (hunk 5 above)
Comment (SandraPG85) [inline VAS.Services/Controller/EventsController.cs]: adding the handleEditevent function

File: VAS.Core/ViewModel/TimelineVM.cs [hunk 7]
```diff
@@ -54,6 +55,7 @@ public TimelineVM ()
 			FullTimeline = CreateFullTimeline ();
 			FullTimeline.ViewModels.CollectionChanged += HandleTimelineCollectionChanged;
 			FullTimeline.PropertyChanged += FullTimeline_PropertyChanged;
+			EditionCommand = new Command<TimelineEvent> (HandleEditPlay, (arg) => true);
```
Comment (SandraPG85) [inline VAS.Core/ViewModel/TimelineVM.cs]: there is a task to fix the tree view behavior, right know the command is only being used by the execute function, so this can be checked there, right now my answer will be that yes :)

File: VAS.Core/Common/StateController.cs (hunk 2 above)
Comment (vguzmanp) [inline VAS.Core/Common/StateController.cs]: Do the same in all other transitions.

File: VAS.Core/Common/StateController.cs [hunk 8]
```diff
@@ -538,7 +538,9 @@ public override bool Equals (object obj)
 		public async Task<bool> Unload ()
 		{
 			bool result = await ScreenState.UnloadState ();
-			Completion.SetResult (result);
+			if (Completion.Task.Status != TaskStatus.RanToCompletion) {
```
Comment (vguzmanp) [inline VAS.Core/Common/StateController.cs]: In which case is the task already completed?

Comment (vguzmanp) [inline VAS.Core/Common/StateController.cs]: After our chat: Add a `TechnicalDebt` task, and a FIXME for this, as it's hiding an exception
